extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/dashboard/DashboardPageApiService.java

```diff
@@ -0,0 +1,68 @@
+/*
+ *  Copyright (c) 2022 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.ext.wrapper.api.ui.pages.dashboard;
+
+import de.sovity.edc.ext.wrapper.api.common.mappers.AssetMapper;
+import de.sovity.edc.ext.wrapper.api.common.mappers.PolicyMapper;
+import de.sovity.edc.ext.wrapper.api.ui.model.ContractAgreementDirection;
+import de.sovity.edc.ext.wrapper.api.ui.model.DashboardPage;
+import de.sovity.edc.ext.wrapper.api.ui.pages.dashboard.services.DashboardDataFetcher;
+import de.sovity.edc.ext.wrapper.api.ui.pages.transferhistory.TransferProcessStateService;
+import lombok.RequiredArgsConstructor;
+import org.jetbrains.annotations.NotNull;
+
+import static de.sovity.edc.ext.wrapper.api.ui.model.TransferProcessSimplifiedState.ERROR;
+import static de.sovity.edc.ext.wrapper.api.ui.model.TransferProcessSimplifiedState.OK;
+import static de.sovity.edc.ext.wrapper.api.ui.model.TransferProcessSimplifiedState.RUNNING;
+
+@RequiredArgsConstructor
+public class DashboardPageApiService {
+    private final DashboardDataFetcher dashboardDataFetcher;
+    private final AssetMapper assetMapper;
+    private final PolicyMapper policyMapper;
+    private final TransferProcessStateService transferProcessStateService;
+
+    @NotNull
+    public DashboardPage dashboardPage(String connectorEndpoint) {
+
+        var runningTransferProcesses = dashboardDataFetcher.getTransferProcessesAmount().stream()
+                .filter(transferProcess -> transferProcessStateService.getSimplifiedState(transferProcess.getState()).equals(RUNNING))
+                .count();
+
+        var completedTransferProcesses = dashboardDataFetcher.getTransferProcessesAmount().stream()
+                .filter(transferProcess -> transferProcessStateService.getSimplifiedState(transferProcess.getState()).equals(OK))
+                .count();
+
+        var interruptedTransferProcesses = dashboardDataFetcher.getTransferProcessesAmount().stream()
+                .filter(transferProcess -> transferProcessStateService.getSimplifiedState(transferProcess.getState()).equals(ERROR))
+                .count();
```

Comment:
are you fetching the same data 3 times?

better: fetch the list of transfer processes once (as list)

then .stream().filter(...).count() it multiple times